solvers/qndiag.py

```diff
@@ -95,4 +95,4 @@ def _linesearch(D, B, direction, current_loss):
         self.B = B
```

Comment:
`stopping_strategy` has been renamed ` sampling_strategy` and now, the callback does not take any argument (it is retrieved by `get_result`

See here for detailed explanation: https://benchopt.github.io/user_guide/performance_curves.html#using-a-callback